weave-js/src/common/hooks/useWindowSize.ts

```diff
@@ -0,0 +1,67 @@
+import _ from 'lodash';
+import {useLayoutEffect, useState} from 'react';
+import {LARGE_BREAKPOINT, MEDIUM_BREAKPOINT} from '../css/breakpoints.styles';
+
+export type ResponsiveSize = 'small' | 'medium' | 'large';
+
+export interface WindowSize {
+  width: number;
+  height: number;
+  responsiveSize: ResponsiveSize;
+}
+
+export const getWindowResponsiveSize = (width: number): ResponsiveSize => {
+  if (width < MEDIUM_BREAKPOINT) {
+    return 'small';
+  }
+  if (width < LARGE_BREAKPOINT) {
+    return 'medium';
+  }
+  return 'large';
+};
```

Comment:
This stuff for the responsive size is new. The rest is copied directly from [core](https://github.com/wandb/core/blob/c333a694384cfe4684aa2619c412fba9ab15320d/frontends/app/src/util/window.ts).